caddyhttp/httpserver/replacer.go

```diff
@@ -225,12 +225,14 @@ func (r *replacer) getSubstitution(key string) string {
 		}
 	}
 	// search response headers then
-	if key[1] == '<' {
-		want := key[2 : len(key)-1]
-		for key, values := range r.responseRecorder.Header() {
-			// Header placeholders (case-insensitive)
-			if strings.EqualFold(key, want) {
-				return strings.Join(values, ",")
+	if r.responseRecorder != nil {
+		if key[1] == '<' {
```

Comment:
These two if statements can be combined with && so we move out one level of nesting. Sorry I didn't clearly suggest that before. :)